trieste/acquisition/function.py

```diff
@@ -509,6 +512,155 @@ def prepare_acquisition_function(
 
 
+class ExpectedHypervolumeImprovement(SingleModelAcquisitionBuilder):
+    """
+    Builder for the :func:`expected_hv_improvement` acquisition function.
+    The implementation of the acquisition function largely
+    follows :cite:`yang2019efficient`
+    """
+
+    def __repr__(self) -> str:
+        """"""
+        return "ExpectedHypervolumeImprovement()"
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ProbabilisticModel
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: The data from the observer. Must be populated.
+        :param model: The model over the specified ``dataset``.
+        :return: The expected_hv_improvement function.
+        """
+        tf.debugging.assert_positive(len(dataset), message="Dataset must be populated.")
+        mean, _ = model.predict(dataset.query_points)
+
+        _pf = Pareto(mean)
+        _reference_pt = get_reference_point(_pf.front)
+        return expected_hv_improvement(model, _pf, _reference_pt)
+
+
+def expected_hv_improvement(
+    model: ProbabilisticModel,
+    pareto: Pareto,
+    reference_point: TensorType,
+) -> AcquisitionFunction:
+    r"""
+    HV calculation using Eq. 44 of :cite:`yang2019efficient` paper.
+    The expected hypervolume improvement calculation in the non-dominated region
+    can be decomposed into sub-calculations based on each partitioned cell.
+    This sub-scalculation can be further reformulated as a combination of two types of
+    generalized expected improvements for easier math operation, corresponding to Psi
+    function and nu function calculations.
+    Mainly referring Eq. 44, and Eq. 45
+    Note:
+    1. Since in Trieste we do not assume the use of a certain non-dominated partition algorithm,
+       we do not assume the last dimension partitioned cell has only one (lower) bound
+       (which is used in the :cite:`yang2019efficient` paper). This is not as efficient as
+        the original paper, but is applicable to different non-dominated partition algorithm.
+    2. The Psi and nu function in the original paper is defined for a maximization problem. To
+       make use of the same notation for easier following, we inverse our problem (as maximization)
+       to make use of the same equation.
+
+    :param model: The model of the objective function.
+    :param pareto: Pareto class
+    :param reference_point The reference point for calculating hypervolume
+    :return The expected_hv_improvement acquisition function modified for objective
+        minimisation. This function will raise :exc:`ValueError` or
+        :exc:`~tf.errors.InvalidArgumentError` if used with a batch size greater than one.
+    """
+
+    def acquisition(x: TensorType) -> TensorType:
+        tf.debugging.assert_shapes(
+            [(x, [..., 1, None])],
+            message="This acquisition function only supports batch sizes of one.",
+        )
```

Comment:
For now ;)

Reply:
Is there any plan to make one acq for EHVI with arbitrary batch size? 
We have a qehvi in prepare indeed, but it is written as another acquisition function.

Reply:
I guess we might merge the two, but thats to discuss another time.